### ISSUE TITLE
lib/Mandrill/Mandrill.php Simply doesn't work and never will as is

### ISSUE THREAD

**Issue by viable-hartman** [mailchimp/mc-magento#1166]
### Preconditions
My environment doesn't matter, the code is wrong.
1.  But I am running 1.9.4.4

### Steps to reproduce
1. With the Mandrill helper enabled in the configuration
2.  Try to send a shipping notification, it won't work.  It will say "Success," but doesn't actually do anything.

### Expected result
1.  I'd expect an e-mail to actually be sent.

### Actual result
1. "Success" is printed, but the e-mail never goes out.  In fact the curl call simply does a GET with ignored POST FIELDS because it is a GET.  You are trying to do a POST of JSON with a curl library that can't do a POST of JSON.  It's simply not designed to work the way you are trying to make it work...

### Solution
1. You can make it work by modifying a core library file as depicted in my patch, but the way your extension is currently coded, it will never work:
```
diff --git a/app/code/community/Ebizmarts/MailChimp/Helper/Curl.php b/app/code/community/Ebizmarts/MailChimp/Helper/Curl.php
index 8b15b248f..6700572ff 100644
--- a/app/code/community/Ebizmarts/MailChimp/Helper/Curl.php
+++ b/app/code/community/Ebizmarts/MailChimp/Helper/Curl.php
@@ -45,6 +45,8 @@ class Ebizmarts_MailChimp_Helper_Curl extends Mage_Core_Helper_Abstract
                 $curl->get($url);
             } elseif ($httpMethod == Zend_Http_Client::POST) {
                 $curl->post($url, $params);
+            } elseif ($httpMethod == 'JSON') {
+                $curl->post_json($url, $params);
             }

             $curlResult = $curl->getBody();

diff --git a/lib/Mage/HTTP/Client/Curl.php b/lib/Mage/HTTP/Client/Curl.php
index 4aa97a9c4..ce6d97d82 100644
--- a/lib/Mage/HTTP/Client/Curl.php
+++ b/lib/Mage/HTTP/Client/Curl.php
@@ -248,6 +248,15 @@ implements Mage_HTTP_IClient
         $this->makeRequest("POST", $uri, $params);
     }

+    /**
+     * Make JSON POST request
+     * @see lib/Mage/HTTP/Mage_HTTP_Client#post_json($uri, $params)
+     */
+    public function post_json($uri, $params)
+    {
+        $this->makeRequest("JSON", $uri, $params);
+    }
+

     /**
      * Get response headers
@@ -357,6 +366,10 @@ implements Mage_HTTP_IClient
             $this->curlOption(CURLOPT_POSTFIELDS, http_build_query($params));
         } elseif($method == "GET") {
             $this->curlOption(CURLOPT_HTTPGET, 1);
+        } elseif($method == "JSON") {
+            $this->curlOption(CURLOPT_CUSTOMREQUEST, 'POST');
+            $this->curlOption(CURLOPT_POSTFIELDS, json_encode($params));
+            $this->curlOption(CURLOPT_HTTPHEADER, array('Content-Type: application/json'));
         } else {
             $this->curlOption(CURLOPT_CUSTOMREQUEST, $method);
         }

diff --git a/lib/Mandrill/Mandrill.php b/lib/Mandrill/Mandrill.php
index e47c3e6de..4ab8f1397 100755
--- a/lib/Mandrill/Mandrill.php
+++ b/lib/Mandrill/Mandrill.php
@@ -139,16 +139,13 @@ class Mandrill_Mandrill
     public function call($url, $params)
     {
         $params['key'] = $this->apikey;
-        $params = json_encode($params);

         $curlOptions = array();
-        $curlOptions[CURLOPT_URL] = $this->root . $url . '.json';
-        $curlOptions[CURLOPT_HTTPHEADER] = array('Content-Type: application/json');
-        $curlOptions[CURLOPT_POSTFIELDS] = $params;
+        $url = $this->root . $url . '.json';
         $curlOptions[CURLOPT_VERBOSE] = $this->debug;

         $start = microtime(true);
-        $this->log('Call to ' . $this->root . $url . '.json: ' . $params);
+        $this->log('Call to ' . $url . ': ' .print_r($params, true));

         if ($this->debug) {
             $curlBuffer = fopen('php://memory', 'w+');
@@ -160,8 +157,7 @@ class Mandrill_Mandrill
          */
         $curlHelper = Mage::helper('mailchimp/curl');
         $this->setCurlOptionsAddOptions($curlOptions);
-        //$curlFullResponse = $curlHelper->curlExec($this->_curlOptions);
-        $curlFullResponse = $curlHelper->curlExec($curlOptions[CURLOPT_URL], Zend_Http_Client::POST, $this->_curlOptions);
+        $curlFullResponse = $curlHelper->curlExec($url, 'JSON', $this->_curlOptions, $params);
         $info = $curlFullResponse['info'];
         $responseBody = $curlFullResponse['response'];
         $curlError = $curlFullResponse['error'];
```

**Comment by viable-hartman** [comment]
I should also note, the $info variable in this lib file is also pointless and will simply just throw warnings in the system log as long as it exists.  Since you are using the Mage/HTTP/Client/Curl client through your MailChimp Curl helper, you will not be able to access the curl info.  Nor should you care to since errors are handled in the curl client class.  Thus, you should also just remove the info and info try catch stuff.

**Comment by Santiagoebizmarts** [comment]
Hello @viable-hartman 

Thanks for your feedback.
Please note this fix is already available when downloading the [latest version 1.1.20](https://github.com/mailchimp/mc-magento/releases/tag/1.1.20).

Kind Regards.
